src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/models/RadPatch.java

```diff
@@ -1,45 +1,125 @@
 package network.radicle.jetbrains.radiclejetbrainsplugin.models;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import git4idea.repo.GitRepository;
 
 import java.time.Instant;
 import java.util.List;
 
 public class RadPatch {
     public GitRepository repo;
-    public String patchId;
-    public String author;
+    public String id;
     public String title;
+    public Author author;
     public String description;
-    public State state;
     public String target;
+    public List<String> tags;
+    public State state;
     public List<Revision> revisions;
 
-    public RadPatch(GitRepository repo, String patchId, String author, String title, String description, State state, String target,
-                    List<Revision> revisions) {
-        this.repo = repo;
-        this.patchId = patchId;
-        this.author = author;
+    public record Author(String id) {
```

Comment:
please move all nested records/enums to the bottom of the class, so they're out of the way when looking at the class fields/methods